packages/ra-auth-cognito/src/MfaTotpAssociationForm.tsx

```diff
@@ -0,0 +1,97 @@
+import { Box, Button, CardContent, CircularProgress } from '@mui/material';
+import clsx from 'clsx';
+import { QRCodeSVG } from 'qrcode.react';
+import * as React from 'react';
+import {
+    Form,
+    LoginFormClasses,
+    TextInput,
+    required,
+    useTranslate,
+} from 'react-admin';
+import type { SubmitHandler } from 'react-hook-form';
+import type { FormData } from './useCognitoLogin';
+
+export type MfaTotpAssociationFormProps = {
+    className?: string;
+    submit: SubmitHandler<FormData>;
+    isLoading?: boolean;
+    secretCode: string;
+    username: string;
+    applicationName: string;
+};
+
+export const MfaTotpAssociationForm = ({
+    className,
+    submit,
+    isLoading,
+    secretCode,
+    username,
+    applicationName,
+}: MfaTotpAssociationFormProps) => {
+    const translate = useTranslate();
+    const otpUrl = `otpauth://totp/${username}?secret=${secretCode}&issuer=${applicationName}`;
+
+    return (
+        <Form
+            onSubmit={submit}
+            mode="onChange"
+            noValidate
+            className={clsx('RaLoginForm-root', className)}
+        >
+            <CardContent className={LoginFormClasses.content}>
+                <Box sx={{ maxWidth: '720px', pb: 2, margin: 'auto' }}>
+                    {translate('ra.auth.cognito.mfa.totp.association_required')}
+                </Box>
+                <Box
+                    sx={{
+                        display: 'flex',
+                        flexDirection: 'column',
+                        alignItems: 'center',
+                        pb: 2,
+                    }}
+                >
+                    <QRCodeSVG value={otpUrl} />
+                    <Button
+                        size="small"
+                        onClick={() => navigator.clipboard.writeText(otpUrl)}
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏Some apps (like Google Authenticator) do not support the full URL in that field. The most compatible way is to copy the secret only.

```suggestion
                        onClick={() => navigator.clipboard.writeText(secretCode)}
```